src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -57,3 +78,252 @@ def get_stats(src_path, dst_path, fs_handle, human=True):
+        # in case this remote call on RTimer fails, its details won't printed,
+        # logged or would cause a crash. therefore, leaving some info for
+        # debugging in logs
+        log.info(f'ev_id = {ev_id} ev_progress_fraction = {ev_progress_fraction}')
+        log.info(f'ev_msg = {ev_msg}')
+
+        self.volclient.mgr.remote('progress', 'update', ev_id=ev_id,
+                                  ev_msg=ev_msg,
+                                  ev_progress=ev_progress_fraction,
+                                  refs=['mds', 'clone'], add_to_ceph_s=True)
+
+    def _update_onpen_progress_bar(self, clones, total_clones, sum_percent):
+        '''
+        Update the progress bar for ongoing + pending cloning operations.
+        '''
+        log.info('updating progress bars for clone jobs')
+        assert self.onpen_pev_id is not None
+
+        total_clones = len(clones)
+        percent = 0.0
+        avg_percent = 0.0
+
+        for clone in clones[total_clones:]:
+            with open_volume_lockless(self.volclient, clone.volname) as \
+                    fs_handle:
+                percent = get_percent_copied(clone.src_path, clone.dst_path,
+                                             fs_handle)
+                sum_percent += percent
+
+        avg_percent = round(sum_percent / total_clones, 3)
+        # progress module takes progress as a fraction between 0.0 to 1.0.
+        avg_progress_fraction = avg_percent / 100
+        msg = (f'{total_clones} ongoing+pending clones; Avg progress is '
+               f'{avg_percent}%')
+        self._update_progress_bar_event(ev_id=self.onpen_pev_id, ev_msg=msg,
+                                    ev_progress_fraction=avg_progress_fraction)
+
+        log.info('finished updating progress bar for ongong+pending clones')
+
+    def _update_ongoing_progress_bar(self, clones):
+        '''
+        Update the progress bar for ongoing cloning operations.
+        '''
+        log.info('updating progress bars for clone jobs')
+        assert self.on_pev_id is not None
+
+        percent = 0.0
+        sum_percent = 0.0
+        avg_percent = 0.0
+
+        if len(clones) < self.max_concurrent_clones:
+            total_clones = len(clones)
+        else:
+            total_clones = self.max_concurrent_clones
+
+        for clone in clones[:total_clones]:
+            with open_volume_lockless(self.volclient, clone.volname) as \
+                    fs_handle:
+                percent = get_percent_copied(clone.src_path, clone.dst_path,
+                                             fs_handle)
+                sum_percent += percent
+
+        avg_percent = round(sum_percent / total_clones, 3)
+        # progress module takes progress as a fraction between 0.0 to 1.0.
+        avg_progress_fraction = avg_percent / 100
+        msg = f'{total_clones} ongoing clones; Avg progress is {avg_percent}%'
+        self._update_progress_bar_event(ev_id=self.on_pev_id, ev_msg=msg,
```

Comment:
No need for a separate helper for this. Just call the remote method directly.

Reply:
I find keeping this entire block of code under a layer of abstraction to be better as it gives to many details that are not necessarily needed for 99% times.

Also this methods looks like this now, so it's even better to move this to a different function -

```
        log.info(f'ev_id = {ev_id} ev_progress_fraction = {ev_progress_fraction}')
        log.info(f'ev_msg = {ev_msg}')
        log.info('calling update() from mgr/update module')

        self.volclient.mgr.remote('progress', 'update', ev_id=ev_id,
                                  ev_msg=ev_msg,
                                  ev_progress=ev_progress_fraction,
                                  refs=['mds', 'clone'], add_to_ceph_s=True)

        log.info('call to update() from mgr/update module was successful')
```